src/routing.ts

```diff
@@ -56,11 +57,26 @@ export const initRouting = ({
       }
       return endpoint.execute({ request, response, logger, config });
     };
-    if (config.cors && !corsedPaths.has(path)) {
-      app.options(path, ...matchingParsers, handler);
-      corsedPaths.add(path);
+    if (!familiar.has(path)) {
+      familiar.set(path, []);
+      if (config.cors) {
+        app.options(path, ...matchingParsers, handler);
+        familiar.get(path)?.push("options");
+      }
     }
+    familiar.get(path)?.push(method);
     app[method](path, ...matchingParsers, handler);
   };
   walkRouting({ routing, onEndpoint, onStatic: app.use.bind(app) });
+  if (config.wrongMethodBehavior !== 405) return;
+  for (const [path, allowedMethods] of familiar.entries()) {
+    app.all(path, ({ method }, res, next) => {
+      return next(
```

Comment:
this is handled by `createParserFailureHandler`.
perhaps it should be generalized